targets/arm/mikroe/sam/src/tim/implementations/implementation_1/hal_ll_tim.c

```diff
@@ -0,0 +1,752 @@
+        set_reg_bits( &hal_ll_hw_reg->channel[ channel_num ].cmr, HAL_LL_TC_CMR_ACPC_SET_MASK );
+    } else {
+        // Clear TIOBx pin on RB compare and set on RC compare. Also set TIOBx as output.
+        set_reg_bits( &hal_ll_hw_reg->channel[ channel_num ].cmr, HAL_LL_TC_CMR_BCPB_CLEAR_MASK );
+        set_reg_bits( &hal_ll_hw_reg->channel[ channel_num ].cmr, HAL_LL_TC_CMR_BCPC_SET_MASK );
+        set_reg_bits( &hal_ll_hw_reg->channel[ channel_num ].cmr, HAL_LL_TC_CMR_EEVT_XC0_MASK );
+    }
+
+    // Enable counter.
+    set_reg_bit( &hal_ll_hw_reg->channel[ channel_num ].ccr, HAL_LL_TC_CCR_CLKEN_BIT );
+
+    // Trigger the counter.
+    set_reg_bit( &hal_ll_hw_reg->channel[ channel_num ].ccr, HAL_LL_TC_CCR_SWTRG_BIT );
+
+    return HAL_LL_TIM_SUCCESS;
+}
+
+hal_ll_err_t hal_ll_tim_stop( handle_t *handle ) {
+    uint8_t channel_num = hal_ll_tim_hw_specifics_map_local->config.channel;
+
+    low_level_handle = hal_ll_tim_get_handle;
+    hal_ll_tim_hw_specifics_map_local = hal_ll_get_specifics( hal_ll_tim_get_module_state_address );
+
+    hal_ll_tim_base_handle_t *hal_ll_hw_reg = hal_ll_tim_get_base_struct( hal_ll_tim_hw_specifics_map_local->base );
+
+    // Disable counter.
+    set_reg_bit( &hal_ll_hw_reg->channel[ channel_num ].ccr, HAL_LL_TC_CCR_CLKDIS_BIT );
+
+    return HAL_LL_TIM_SUCCESS;
+}
+
+void hal_ll_tim_close( handle_t *handle ) {
+
+    low_level_handle = hal_ll_tim_get_handle;
+    hal_ll_tim_hw_specifics_map_local = hal_ll_get_specifics(hal_ll_tim_get_module_state_address);
+
+    if( low_level_handle->hal_ll_tim_handle != NULL ) {
+        low_level_handle->hal_ll_tim_handle = NULL;
+        low_level_handle->hal_drv_tim_handle = NULL;
+
+        low_level_handle->init_ll_state = false;
+
+        hal_ll_tim_hw_specifics_map_local->max_period = 0;
+        hal_ll_tim_hw_specifics_map_local->freq_hz = 0;
+
+        _hal_ll_tim_set_clock( hal_ll_tim_hw_specifics_map_local->base, true );
+        _hal_ll_tim_alternate_functions_set_state( hal_ll_tim_hw_specifics_map_local, false );
+        _hal_ll_tim_set_clock( hal_ll_tim_hw_specifics_map_local->base, false );
+
+        hal_ll_tim_hw_specifics_map_local->config.pin = HAL_LL_PIN_NC;
+        hal_ll_tim_hw_specifics_map_local->config.channel = HAL_LL_PIN_NC;
+        hal_ll_tim_hw_specifics_map_local->config.af = NULL;
+    }
+}
+
+// ----------------------------------------------- PRIVATE FUNCTION DEFINITIONS
+static inline void _hal_ll_tim0_set_clock( hal_ll_channel_t channel_num, bool hal_ll_state ) {
+    uint8_t id_pos;
+
+    // Get the appropriate PID number.
+    switch ( channel_num )
+    {
+        case HAL_LL_TIM_CHANNEL_0:
+            id_pos = HAL_LL_PID_TC0_BIT;
+            break;
+        case HAL_LL_TIM_CHANNEL_1:
+            id_pos = HAL_LL_PID_TC1_BIT;
+            break;
+        case HAL_LL_TIM_CHANNEL_2:
+            id_pos = HAL_LL_PID_TC2_BIT;
+            break;
+    }
```

Comment:
Default case?

Reply:
Added it for all source files where needed.

Reply:
No default case again.

Reply:
Added id_pos default value to be like HAL_LL_TIM_MODULE_ERROR